mix.exs

```diff
@@ -48,9 +48,11 @@ defmodule App.MixProject do
     [
       {:ex_doc, "~> 0.19.0", only: :dev},
       {:excoveralls, "~> 0.10", only: :test},
-      {:ecto, "~> 2.2.0", only: [:dev, :test]},
       {:credo, "~> 0.10.0", only: [:dev, :test]},
-      {:dialyxir, "~> 1.0.0-rc.4", only: [:dev], runtime: false}
+      {:dialyxir, "~> 1.0.0-rc.4", only: [:dev], runtime: false},
+
+      # optional
+      {:ecto, "~> 2.2.0", optional: true}
```

Comment:
Why not use Ecto 3.0?

Reply:
I haven't used `Ecto 3.x` yet, so I can't say for sure it's compatible with `2.x` on this validations.
Can you test it?

Reply:
No have breaking change in validation: http://blog.plataformatec.com.br/2018/10/a-sneak-peek-at-ecto-3-0-breaking-changes/

Reply:
Nice. to make it compatible w/ `ecto 2.x`, I'll add this dependency as `"~> 2.2 or ~> 3.0"`